source/AI.cpp

```diff
@@ -1594,7 +1581,8 @@ void AI::MoveEscort(Ship &ship, Command &command) const
 				ship.SetTargetStellar(nullptr);
 		}
 
-		if(!ship.GetTargetStellar() && !ship.GetTargetSystem())
+		// If the ship has no destination or the destination is unreachable, route to the parent's system.
+		if(!ship.GetTargetStellar() && (!ship.GetTargetSystem() || !ship.JumpFuel(ship.GetTargetSystem())))
```

Comment:
The only other comment I have is here. If Ship::JumpFuel isn't improved then this should probably be changed to the following.
```suggestion
		// Since the JumpFuel calculation can be expensive, only check twice per second.
		if(!ship.GetTargetStellar() && (!ship.GetTargetSystem() || !Random::Int(30) && !ship.JumpFuel(ship.GetTargetSystem())))
```
But should we merge #7378, this won't be an issue.

Reply:
I notice no significant decrease in performance idling in Sol with this change, so I think we're safe to keep the existing JumpFuel without needing the random call.